elide-core/src/main/java/com/yahoo/elide/parsers/expression/PermissionToFilterExpressionVisitor.java

```diff
@@ -157,4 +163,42 @@ private Operator operator(FilterExpression expression) {
     public FilterExpression visitPAREN(ExpressionParser.PARENContext ctx) {
         return visit(ctx.expression());
     }
+
+    /**
+     * normalizeNegation function serves to push down NotFilterexpression to the predicate level.
+     * @param fe FilterExpression that need to normalize
+     * @return A normalized version of input
+     */
+    public static FilterExpression normalizeNegation(FilterExpression fe) {
```

Comment:
I'm having a hard time understanding why we make this a `public static` in this class. It doesn't seem like this class needs to ever perform this operation. Should we move this to the `PermissionExpressionBuilder` instead?